helm-chart/binderhub/templates/deployment.yaml

```diff
@@ -62,6 +62,27 @@ spec:
             secretKeyRef:
               name: binder-secret
               key: "binder.hub-token"
+        {{ if .Values.github.clientId -}}
```

Comment:
While this is ok for now, I think in the long run we should load the secret as a volume and read these from binderhub_config.py. Just  to keep the deployment definition clean.